source/PlayerInfo.cpp

```diff
@@ -4142,6 +4143,15 @@ void PlayerInfo::StepMissions(UI *ui)
 }
 
 
+void PlayerInfo::StepMissionTimers(UI *ui)
+{
+	for(Mission &mission : missions)
+		for(Timer &timer : mission.Timers())
+			if(!timer.IsComplete())
+				timer.Step(*this, ui);
```

Comment:
Step should be function to check whether the timer has completed.

Reply:
....So you'd want `Step()` to just have right at the start of it a short-circuit check for completion, in which case it returns a boolean? (Not sure whether `true` or `false` would make most sense...)

What, exactly, would that boolean value be for at that point? Would it be just as good simply to move the completion check into `Step()` and _not_ have it return a value?

Reply:
> So you'd want Step() to just have right at the start of it a short-circuit check for completion

It already has one :) No need to return anything either.